syntax ="proto3";

message requestType {
    int32 type = 1;
}

message listarAlunosRequest {
    string codigoDisciplina = 1;
    int32 ano = 2;
    int32 semestre = 3;
}

message listarAlunosResponse {
    message Aluno{
        int32 ra = 1;
        string nome = 2;
        int32 periodo = 4;
    }
    repeated Aluno alunos = 1;
    string mensagem = 2;
}

message alterarNotaRequest {
    int32 ra = 1;
    string codigoDisciplina = 2;
    int32 ano = 3;
    int32 semestre = 4;
    float nota = 5;
}

message alterarNotaResponse {
    int32 ra = 1;
    string codigoDisciplina = 2;
    int32 ano = 3;
    int32 semestre = 4;
    float nota = 5;
    string mensagem = 7;
}

message alterarFaltasRequest {
    int32 ra = 1;
    string codigoDisciplina = 2;
    int32 ano = 3;
    int32 semestre = 4;
    int32 faltas = 5;
}

message alterarFaltasResponse {
    int32 ra = 1;
    string codigoDisciplina = 2;
    int32 ano = 3;
    int32 semestre = 4;
    int32 faltas = 5;    
    string mensagem = 6;
}

message listarDisciplinasAlunoRequest {
    int32 ra = 1;
    int32 ano = 2;
    int32 semestre = 3;
}

message listarDisciplinasAlunoResponse {
    message DisciplinaAlunos{
        int32 ra = 1;
        string codigoDisciplina = 2;
        float nota = 3;
        int32 faltas = 4;  
    }
    repeated DisciplinaAlunos disciplinas = 1;
    string mensagem = 2;
}

message inserirMatriculaRequest {
    Matricula matricula = 1;
}

message Matricula{
    int32 ra = 1;
    string codigoDisciplina = 2;
    int32 ano = 3;
    int32 semestre = 4;
    optional float nota = 5;
    optional int32 faltas = 6;
}

message inserirMatriculaResponse {
    Matricula matricula = 1;
    string mensagem = 2;
}
